src/test/java/io/cryostat/recordings/RecordingTargetHelperTest.java

```diff
@@ -806,6 +807,176 @@ public Future<Metadata> answer(InvocationOnMock invocation)
+                .thenReturn(Optional.of(events));
+
+        Mockito.when(
+                        recordingMetadataManager.setRecordingMetadata(
+                                Mockito.any(), Mockito.anyString(), Mockito.any(Metadata.class)))
+                .thenAnswer(
+                        invocation -> {
+                            return CompletableFuture.completedFuture(invocation.getArgument(2));
+                        });
+
+        Mockito.doNothing().when(notification).send();
+
+        recordingTargetHelper.startRecording(
+                false,
+                "always",
+                connectionDescriptor,
+                recordingOptions,
+                templateName,
+                templateType,
+                metadata,
+                false);
+
+        Mockito.verify(service).start(Mockito.any(), Mockito.any());
+
+        // Verify that the notification is not sent since the recording already exists and no new
+        // recording is created
+        Mockito.verify(notification, Mockito.times(0)).send();
+    }
+
+    @Test
+    void shouldDoNothingWhenRecordingExistsAndIsRunning() throws Exception {
+        String recordingName = "existingRecording";
+        String targetId = "fooTarget";
+        String duration = "5000ms";
+        String templateName = "Profiling";
+        TemplateType templateType = TemplateType.TARGET;
+        ConnectionDescriptor connectionDescriptor = new ConnectionDescriptor(targetId);
+        IRecordingDescriptor existingRecording = createDescriptor(recordingName);
+        IConstrainedMap<String> recordingOptions = Mockito.mock(IConstrainedMap.class);
+        Metadata metadata =
+                new Metadata(Map.of("template.name", "Profiling", "template.type", "TARGET"));
+
+        Mockito.when(targetConnectionManager.executeConnectedTask(Mockito.any(), Mockito.any()))
+                .thenAnswer(
+                        invocation -> {
+                            TargetConnectionManager.ConnectedTask task = invocation.getArgument(1);
+                            return task.execute(connection);
+                        });
+
+        Mockito.when(recordingOptions.get(Mockito.any())).thenReturn(recordingName, duration);
+
+        Mockito.when(connection.getService()).thenReturn(service);
+        List<IRecordingDescriptor> existingRecordings = List.of(existingRecording);
+        Mockito.when(service.getAvailableRecordings()).thenReturn(existingRecordings);
+
+        Mockito.when(existingRecording.getState()).thenReturn(RecordingState.RUNNING);
+
+        recordingTargetHelper.startRecording(
+                false,
+                "always",
+                connectionDescriptor,
+                recordingOptions,
+                templateName,
+                templateType,
+                metadata,
+                false);
+
+        Mockito.verify(service, Mockito.never()).start(Mockito.any(), Mockito.any());
```

Comment:
I think if the parameter is `"always"`, then this should actually mean that the `existingRecording` gets closed and a new recording started in its place, even though it is already running, no?

Reply:
From the comment that @tthvo had originally, nothing should be done if recording exists and running. because it wont capture any additional data. This behavior though contradicts with the word  `always`.

Reply:
Sorry I missed this comment. I think Andrew's comment above is the correct intended behaviour.

The original paragraph (1st one I believe) is talking about how this `replace` parameter can be used within the topology view (i.e. Target detail sidecar). In other words, it was meant to say we might use `replace=stopped` there.